packages/fast-components/src/checkbox/index.ts

```diff
@@ -2,12 +2,29 @@ import { customElement } from "@microsoft/fast-element";
 import { Checkbox } from "./checkbox";
 import { CheckboxTemplate as template } from "./checkbox.template";
 import { CheckboxStyles as styles } from "./checkbox.styles";
+import { designSystemConsumer } from "../design-system-consumer";
+import {
+    neutralfillinputhover,
+    neutralfillinputrest,
+    neutralforegroundrest,
+    neutraloutlinehover,
+    neutraloutlinerest,
+} from "../styles/recipes";
 
 @customElement({
     name: "fast-checkbox",
     template,
     styles,
 })
+@designSystemConsumer({

```

Comment:
If I'm reading this correctly, it looks like I would need to import and associate a recipe like neutralOutlineRest in the component, then I could reference `var(--neutral-outline-rest)` in the styles. Having this disconnected seems difficult to keep in sync, and doesn't seem like it supports swapping styles on existing components. I'm sure I don't know most of the internals of the model, but is there a way to swap or chain up this relationship so the declarative part comes in the `css` function instead of the component?

Reply:
Yea I agree. Part of this was to get *an* implementation up and see where the gaps are and what API changes we might need to make to fast-element to get a good experience with recipes. I don't think we have infrastructure to associate recipes with a stylesheet instance at the moment.

Reply:
To make sure I understand, you would want to associate recipes with a stylesheet, and then when you apply the styles, also apply the recipes. Is that correct? If so, I think we could do that without a ton of trouble. That could get pretty interesting considering also the idea of registering things by name so that an appearance change could add and remove the styles dynamically (and now recipes).

Let me know if my understanding is correct on that.

Reply:
Yep, exactly. Only the stylesheet should *really* care about it's dependent recipes, not the component.

Reply:
I think we can make that happen. So, if we combine this with the registry idea, would we want something like this:

```TypeScript
const styles = css`
  ..some css here..
`.withRecipes([
    neutralOutlineRest,
    neutralFillInputRest,
    neutralForegroundRest,
    neutralFillInputHover,
    neutralOutlineHover,
].register('foo');
```